aspnetcore/migration/claimsprincipal-current.md

```diff
@@ -22,6 +22,14 @@ There are several options for retrieving the current authenticated user's `Claim
 * **ControllerBase.User**. MVC controllers can access the current authenticated user with their <xref:Microsoft.AspNetCore.Mvc.ControllerBase.User%2A> property.
 * **HttpContext.User**. Components with access to the current `HttpContext` (middleware, for example) can get the current user's `ClaimsPrincipal` from <xref:Microsoft.AspNetCore.Http.HttpContext.User%2A?displayProperty=nameWithType>.
 * **Passed in from caller**. Libraries without access to the current `HttpContext` are often called from controllers or middleware components and can have the current user's identity passed as an argument.
+*  **Exposing Scoped service for HttpContext**. Expose a scoped service for HttpContext to retrieve the current user from.Now any component can just declare a constructor dependency and get the right instance.
```

Comment:
This seems redundant with the next suggestion for using IHttpContextAccessor which works even in non-scoped services.